### PR TITLE
Splunk fixes

### PR DESCRIPTION
**Pull request by ayman-m**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

##### SplunkPy
- Fixed an issue with the drilldown enrichment query where "earliest" and "latest" were mismatched in the query pipeline not the filter. 
- ~~Fixed a problem that causes outgoing mirroring to fail incase of username & password authentication~~, fixed in PR #20662
- ~~Fixed a problem that causes splunk-notable-event-edit to fail incase of username & password authentication~~, fixed in PR #20662

##### SplunkShowAsset
- Fixed an issue with the location where the Asset enrichment data is stored under the incident context.

##### SplunkShowDrilldown
- Fixed an issue with the location where the Drilldown enrichment data is stored under the incident context.

##### SplunkShowIdentity
- Fixed an issue with the location where the Identity enrichment data is stored under the incident context.

## Minimum version of Cortex XSOAR
- [x ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [ x] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

**Comment by ShahafBenYakir** [comment]
@ilappe Please be involved here too.

File: Packs/SplunkPy/Scripts/SplunkShowAsset/SplunkShowAsset.py [hunk 1]
```diff
@@ -1,33 +1,26 @@
+register_module_line('SplunkShowAsset', 'start', __line__())
```
Comment (yaakovi) [inline Packs/SplunkPy/Scripts/SplunkShowAsset/SplunkShowAsset.py:1]:
```suggestion
```

File: Packs/SplunkPy/Scripts/SplunkShowAsset/SplunkShowAsset.py [hunk 2]
```diff
@@ -1,33 +1,26 @@
...
 import json
 
-import demistomock as demisto  # noqa: F401
-from CommonServerPython import *  # noqa: F401
-
```
Comment (yaakovi) [inline Packs/SplunkPy/Scripts/SplunkShowAsset/SplunkShowAsset.py:5]: restore

File: Packs/SplunkPy/Scripts/SplunkShowAsset/SplunkShowAsset.py [hunk 3]
```diff
@@ -37,3 +30,5 @@ def main():
         return_results(main())
     except Exception as e:
         return_error(f'Got an error while parsing Splunk events: {e}', error=e)
+
+register_module_line('SplunkShowAsset', 'end', __line__())
```
Comment (yaakovi) [inline Packs/SplunkPy/Scripts/SplunkShowAsset/SplunkShowAsset.py:34]:
```suggestion
```

File: Packs/SplunkPy/Scripts/SplunkShowDrilldown/SplunkShowDrilldown.py [hunk 4]
```diff
@@ -1,33 +1,26 @@
+register_module_line('SplunkShowDrilldown', 'start', __line__())
```
Comment (yaakovi) [inline Packs/SplunkPy/Scripts/SplunkShowDrilldown/SplunkShowDrilldown.py:1]:
```suggestion
```

File: Packs/SplunkPy/Scripts/SplunkShowDrilldown/SplunkShowDrilldown.py [hunk 5]
```diff
@@ -1,33 +1,26 @@
...
 import json
 
-import demistomock as demisto  # noqa: F401
-from CommonServerPython import *  # noqa: F401
-
```
Comment (yaakovi) [inline Packs/SplunkPy/Scripts/SplunkShowDrilldown/SplunkShowDrilldown.py:5]: restore

File: Packs/SplunkPy/Scripts/SplunkShowDrilldown/SplunkShowDrilldown.py [hunk 6]
```diff
@@ -37,3 +30,5 @@ def main():
         return_results(main())
     except Exception as e:
         return_error(f'Got an error while parsing Splunk events: {e}', error=e)
+
+register_module_line('SplunkShowDrilldown', 'end', __line__())
```
Comment (yaakovi) [inline Packs/SplunkPy/Scripts/SplunkShowDrilldown/SplunkShowDrilldown.py:34]:
```suggestion
```

File: Packs/SplunkPy/Scripts/SplunkShowIdentity/SplunkShowIdentity.py [hunk 7]
```diff
@@ -1,33 +1,26 @@
+register_module_line('SplunkShowIdentity', 'start', __line__())
```
Comment (yaakovi) [inline Packs/SplunkPy/Scripts/SplunkShowIdentity/SplunkShowIdentity.py:1]:
```suggestion
```

File: Packs/SplunkPy/Scripts/SplunkShowIdentity/SplunkShowIdentity.py [hunk 8]
```diff
@@ -1,33 +1,26 @@
...
 import json
 
-import demistomock as demisto  # noqa: F401
-from CommonServerPython import *  # noqa: F401
-
```
Comment (yaakovi) [inline Packs/SplunkPy/Scripts/SplunkShowIdentity/SplunkShowIdentity.py:5]: restore

File: Packs/SplunkPy/Scripts/SplunkShowIdentity/SplunkShowIdentity.py [hunk 9]
```diff
@@ -37,3 +30,5 @@ def main():
         return_results(main())
     except Exception as e:
         return_error(f'Got an error while parsing Splunk events: {e}', error=e)
+
+register_module_line('SplunkShowIdentity', 'end', __line__())
```
Comment (yaakovi) [inline Packs/SplunkPy/Scripts/SplunkShowIdentity/SplunkShowIdentity.py:34]:
```suggestion
```

File: Packs/SplunkPy/Scripts/SplunkShowAsset/SplunkShowAsset.py (hunk 2 above)
Comment (ayman-m) [reply Packs/SplunkPy/Scripts/SplunkShowAsset/SplunkShowAsset.py:5]: @yaakovi restore?!

**Comment by bziser** [comment]
I see we have here duplication in our PRs:
https://github.com/demisto/content/pull/20554

@yaakovi @ilappe

**Comment by ayman-m** [comment]
I just have pushed another commit to rollback the latest docker image update, docker image 1.0.0.32879 uses splunk-sdk 1.6.4 which cause token based authentication to fail

**Comment by bziser** [comment]
@ayman-m Why did you remove the fix of the location where the Drilldown enrichment data is stored under the incident context?

File: Packs/SplunkPy/Scripts/SplunkShowAsset/SplunkShowAsset.py (hunk 2 above)
Comment (yaakovi) [reply Packs/SplunkPy/Scripts/SplunkShowAsset/SplunkShowAsset.py:5]: Add those imports back.

File: Packs/SplunkPy/ReleaseNotes/2_4_10.md [hunk 10]
```diff
@@ -0,0 +1,18 @@
+
+#### Integrations
+##### SplunkPy
+- Fixed an issue with the drilldown enrichment query where "earliest" and "latest" were mismatched in the query pipeline not the filter. 
+- Fixed an issue with the outgoing mirroring where it fail incase the splunk integration is configured with username & password authentication.
+- Fixed an issue with splunk-notable-event-edit where it fail incase the splunk integration is configured with username & password authentication. 
+- Fixed an issue with token based authentication because of an out-dated splunk sdk.
+- Updated the Docker image to: demisto/splunksdk:1.0.0.24033
```
Comment (yuvalbenshalom) [inline Packs/SplunkPy/ReleaseNotes/2_4_10.md:8]:
```suggestion
```

File: Packs/SplunkPy/Integrations/SplunkPy/SplunkPy.py [hunk 11]
```diff
@@ -2633,7 +2633,6 @@ def main():
         connection_args['username'] = username
         connection_args['password'] = password
         connection_args['autologin'] = True
```
Comment (yuvalbenshalom) [inline Packs/SplunkPy/Integrations/SplunkPy/SplunkPy.py:2635]:
```suggestion
        connection_args['autologin'] = True
        connection_args['basic'] = True
```
Please do not remove basic. this issue was handled in a better way in https://github.com/demisto/content/pull/20662

**Review by yaakovi — APPROVED** [review]
I don't have write permissions on the forked repo, I will add fixes in internal PR.

**Pull request merged**